exporter/dynatraceexporter/serialization/serialization.go

```diff
@@ -0,0 +1,178 @@
+	output := ""
+	for i := 0; i < data.Len(); i++ {
+		p := data.At(i)
+		if p.IsNil() {
+			continue
+		}
+
+		tagline := serializeTags(p.LabelsMap(), tags)
+		valueLine := strconv.FormatInt(p.Value(), 10)
+
+		output += serializeLine(name, tagline, valueLine, p.Timestamp())
+	}
+
+	return output
+}
+
+// SerializeDoubleDataPoints serializes a slice of double datapoints to a Dynatrace gauge.
+func SerializeDoubleDataPoints(name string, data pdata.DoubleDataPointSlice, tags []string) string {
+	output := ""
+	for i := 0; i < data.Len(); i++ {
+		p := data.At(i)
+		if p.IsNil() {
+			continue
+		}
+
+		output += serializeLine(name, serializeTags(p.LabelsMap(), tags), serializeFloat64(p.Value()), p.Timestamp())
+	}
+
+	return output
+}
+
+// SerializeDoubleHistogramMetrics serializes a slice of double histogram datapoints to a Dynatrace gauge.
+//
+// IMPORTANT: Min and max are required by Dynatrace but not provided by histogram so they are assumed to be the average.
+func SerializeDoubleHistogramMetrics(name string, data pdata.DoubleHistogramDataPointSlice, tags []string) string {
+	output := ""
+	for i := 0; i < data.Len(); i++ {
+		p := data.At(i)
+		if p.IsNil() {
+			continue
+		}
+
+		tagline := serializeTags(p.LabelsMap(), tags)
+		count := float64(p.Count())
+		avg := p.Sum() / count
+
+		valueLine := fmt.Sprintf("gauge,min=%[1]s,max=%[1]s,sum=%s,count=%d", serializeFloat64(avg), serializeFloat64(p.Sum()), p.Count())
+
+		output += serializeLine(name, tagline, valueLine, p.Timestamp())
+	}
+
+	return output
+}
+
+// SerializeIntHistogramMetrics serializes a slice of integer histogram datapoints to a Dynatrace gauge.
+//
+// IMPORTANT: Min and max are required by Dynatrace but not provided by histogram so they are assumed to be the average.
+func SerializeIntHistogramMetrics(name string, data pdata.IntHistogramDataPointSlice, tags []string) string {
+	output := ""
+	for i := 0; i < data.Len(); i++ {
+		p := data.At(i)
+		if p.IsNil() {
+			continue
+		}
+
+		tagline := serializeTags(p.LabelsMap(), tags)
+
+		avg := float64(p.Sum()) / float64(p.Count())
+
+		valueLine := fmt.Sprintf("gauge,min=%[1]s,max=%[1]s,sum=%d,count=%d", serializeFloat64(avg), p.Sum(), p.Count())
+
+		output += serializeLine(name, tagline, valueLine, p.Timestamp())
+	}
+
+	return output
+}
+
+func serializeLine(name, tagline, valueline string, timestamp pdata.TimestampUnixNano) string {
+	output := name
+
+	if tagline != "" {
+		output += "," + tagline
+	}
+
+	tsMilli := timestamp / 1000
+
+	output += " " + valueline + " " + strconv.FormatUint(uint64(tsMilli), 10) + "\n"
+
+	return output
+}
+
+func serializeTags(labels pdata.StringMap, exporterTags []string) string {
+	tags := append([]string{}, exporterTags...)
+	labels.ForEach(func(k string, v string) {
+		key, err := NormalizeString(strings.ToLower(k), maxDimKeyLen)
+		if err != nil {
+			return
+		}
+		tag := key + "=" + v
```

Comment:
Assuming it's safe to not have any escaping or anything of the value?

Reply:
Added escaping but I think the characters requiring escaping are not allowed by the spec anyways